tools/data-handler/src/interfaces/resource-interfaces.ts

```diff
@@ -0,0 +1,148 @@
+/**
+    Cyberismo
+    Copyright © Cyberismo Ltd and contributors 2024
+
+    This program is free software: you can redistribute it and/or modify it under the terms of the GNU Affero General Public License version 3 as published by the Free Software Foundation.
+
+    This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more details.
+
+    You should have received a copy of the GNU Affero General Public
+    License along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+import { Schema } from 'jsonschema';
+
+/**
+ * Each resource represents a file (or a folder in some cases) with metadata stored
+ * in JSON file. Name of the file is the name of the resource. Each resource is expected to be
+ * in a correct place in folder structure.
+ */
+
+// Calculation does not have own metadata.
+export type CalculationMetadata = ResourceBaseMetadata;
+
+// Card type content.
+export interface CardType extends ResourceBaseMetadata {
+  workflow: string;
+  customFields?: CustomField[];
```

Comment:
Maybe not part of this PR, but we could consider making these required in typescript and add an empty array if the value is missing. It's much easier to work with empty arrays rather than undefined values

Reply:
Yes

Reply:
https://cyberismo.atlassian.net/jira/software/projects/INTDEV/boards/4?selectedIssue=INTDEV-543